app/controllers/application_controller.rb

```diff
@@ -49,36 +49,40 @@ def deliver_sms_to(resource, body_msg)
     if resource.phone_number.blank?
       return "blank"
     end
-    acc_sid = current_user.casa_org.twilio_account_sid
-    api_key = current_user.casa_org.twilio_api_key_sid
-    api_secret = current_user.casa_org.twilio_api_key_secret
+
     body = body_msg
     to = resource.phone_number
     from = current_user.casa_org.twilio_phone_number
 
-    twilio = TwilioService.new(api_key, api_secret, acc_sid)
-    req_params = {
-      From: from,
-      Body: body,
-      To: to
-    }
-
     begin
-      twilio_res = twilio.send_sms(req_params)
+      @twilio = TwilioService.new(current_user.casa_org)
+      req_params = {
+        From: from,
+        Body: body,
+        To: to
+      }
+      twilio_res = @twilio.send_sms(req_params)
       twilio_res.error_code.nil? ? "sent" : "error"
-    rescue Twilio::REST::RestError
+    rescue TwilioService::TwilioCasaOrgError => error
+      @error = error
+      "error"
+    rescue Twilio::REST::RestError => error
+      @error = error
+      "error" # Find a better description for this
+    rescue # unverfied error isnt picked up by Twilio::Rest::RestError
+      # https://www.twilio.com/docs/errors/21608
+      @error = "Phone number is unverifiied"
       "error"
     end
   end
 
-  def sms_acct_creation_notice(resource_name, sms_status)
-    if sms_status === "blank"
-      return "New #{resource_name} created successfully."
-    end
-    if sms_status === "error"
-      return "New #{resource_name} created successfully. SMS not sent due to error."
-    end
-    if sms_status === "sent"
+  def sms_acct_creation_notice(resource_name, sms_status) # switch case implement here (should read mutually exclusive)
```

Comment:
seems reasonable. The comment seems excessive to me but I don't have a strong opinion.

Reply:
Oh my bad that comment should've been scrubbed. It's a note from a pairing session.